core/services/job/orm.go

```diff
@@ -253,18 +254,51 @@ func (o *orm) CreateJob(jb *Job, qopts ...pg.QOpt) error {
 					if !useForwarders && err != nil {
 						return errors.Wrapf(ErrNoSuchTransmitterKey, "%v", jb.OCR2OracleSpec.TransmitterID)
 					}
+
+					newChainID, err := EVMChainIDForJobSpec(jb.OCR2OracleSpec)
+					if err != nil {
+						return err
+					}
+
+					var spec OCR2OracleSpec
+					chainIdPath := []string{"chainID"}
+					err = tx.Get(&spec, `SELECT * FROM ocr2_oracle_specs WHERE relay = $1 AND contract_id = $2 AND relay_config #>> $3 = $4 LIMIT 1`,
+						"EVM", jb.OCR2OracleSpec.ContractID, chainIdPath, fmt.Sprintf("%v", newChainID),
```

Comment:
Doesn't postgres expect: `'{a,b,1}'`?
I'm surprised this works as: `[chainID]`

Reply:
It shouldn't work with `[chainID]`.

The purpose of `Get()` accepting argument parameters, mapped to $ vars instead of just a single sql string is so that it can safely convert native go types into their corresponding sql representations.  It should automatically format the array as `{'chainID'}` (or possibly `ARRAY['chainID']`, depending on how they've implemented it)

Reply:
I'm sorry, I don't understand. `fmt.Sprintf("%v", newChainID)` returns the string `"[chainID]"` which is passed to `Get()`. Are you saying that string is reinterpreted as an array and then formatted again correctly as postgres expects?

Reply:
Oh I misread the variables. If the `fmt.Sprintf("%v"` is just being used to format the chain ID, is it actually necessary?
If we need to make a string, there are some options in `strconv` that might fit the type better, or just `fmt.Sprint(newChainID)` could be simpler.

Reply:
`newChainID.String()`?

Reply:
It is surprisingly an `int64` as you noticed elsewhere. I think the difference is whether the dynamic `JSONConfig` type is used, or a fully typed struct.
https://github.com/smartcontractkit/chainlink/blob/e44008f56b5ecc74b0e342cdb1374ba774c1342d/core/services/job/models.go#L254

Reply:
It would be safer to use `json.Number` for the dynamic ones, to retain the underlying string value precisely: https://pkg.go.dev/encoding/json#Decoder.UseNumber

Reply:
Yes!